src/MyObjectNode.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Created by c4q-Allison on 5/28/15.
+ */
+public abstract class MyObjectNode extends MyNode
+{
+    public MyObjectNode left;
+    public MyObjectNode right;
+    public MyObjectNode data;
+
+    public MyObjectNode()
+    {
+        MyObjectNode myObjectNode = new MyObjectNode()
```

Comment:
MyObjectNode() should override these methods. Very confusing that you're encapsulating an object of its own type.